web/src/views/InvestigationDetail.vue

```diff
@@ -254,9 +259,10 @@ export default defineComponent({
       if (!featureLayer || !pinFeature || !map.value) { return; }
       pinFeature.getData().forEach((pin: object) => {
         if (!pinFeature) return;
-        const { x, y } = postGisToPoint((pin as Pin).location) || { x: 0, y: 0 };
+        const pinObject = pin as Pin;
```

Comment:
There is an improvement to this style of "casting" (which is generally unsafe in this context). You should use a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) here to ensure the values are in fact of the right type (if, instead, it's not possible to type `pinFeature.getData()` as `Pin[]`). Let me know if you want to discuss this further.